subsystems/intakesubsystem.py

```diff
@@ -0,0 +1,122 @@
+
+        self.pivotEncoder = CTREEncoder(
+            constants.kPivotEncoderID,
+            constants.kIntakeAngleOffset.degrees(),
+        )
+
+        self.intakeMotor = NEOBrushless(
+            constants.kIntakeCANID,
+            constants.kIntakeName,
+            constants.kIntakePIDSlot,
+            constants.kIntakePGain,
+            constants.kIntakeIGain,
+            constants.kIntakeDGain,
+            constants.kIntakeInverted,
+        )
+        self.pivotMotor = Talon(
+            constants.kPivotCANID,
+            constants.kPivotName,
+            constants.kPivotPGain,
+            constants.kPivotIGain,
+            constants.kPivotDGain,
+            constants.kPivotInverted,
+        )
+
+        # pivot motor spins 60 times per arm revolution
+        # use absolute encoder to determine motor position
+        pivotMotorPosition = (
+            self.pivotEncoder.getPosition().degrees()
+            / constants.kDegeersPerRevolution
+            * 60
+        )
+        self.pivotMotor.setEncoderPosition(pivotMotorPosition)
+
+        self.state = self.IntakeState.Holding
+
+    def periodic(self) -> None:
+        # get actual velocity values for intake motor later
+
+        if self.state == self.IntakeState.Intaking:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kFloorPositionAngle.degrees()
+                / constants.kDegeersPerRevolution,
+            )
+            if self.intakeMotor.getLimitSwitch(NEOBrushless.LimitSwitch.Forwards):
+                self.intakeMotor.setNeutralOutput(NEOBrushless.NeutralMode.Brake)
+            else:
+                self.intakeMotor.set(NEOBrushless.ControlMode.Velocity, 100)
+
+        elif self.state == self.IntakeState.Holding:
+            self.pivotMotor.set(Talon.ControlMode.Position, 0)
+            self.intakeMotor.setNeutralOutput(NEOBrushless.NeutralMode.Brake)
+
+        elif self.state == self.IntakeState.Feeding:
+            self.pivotMotor.set(Talon.ControlMode.Position, 0)
+            self.intakeMotor.set(NEOBrushless.ControlMode.Velocity, -100)
+
+        elif self.state == self.IntakeState.Staging:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kStagingPositionAngle.degrees()
+                / constants.kDegeersPerRevolution,
+            )
+            self.intakeMotor.setNeutralOutput(NEOBrushless.NeutralMode.Brake)
+
+        elif self.state == self.IntakeState.Amp:
+            self.pivotMotor.set(
+                Talon.ControlMode.Position,
+                constants.kStagingPositionAngle.degrees()
+                / constants.kDegeersPerRevolution,
+            )
+            self.intakeMotor.set(NEOBrushless.ControlMode.Velocity, -100)
+
+        elif self.state == self.IntakeState.Trap:
+            # move with timeout
+            self.pivotMotor.motor.set_position(
```

Comment:
is this a valid function? what does it do?
(more my question than something to fix)

Reply:
Yeah same thing, why not just .set(Talon.ControlMode.Position,)?

Reply:
it adds a timeout, I'm honestly not too sure if/how it will work

Reply:
I'm resolving to get this done faster -- may want to use wpilib Timer to implement this in the future